### DISCUSSION TITLE
Generate OTP for users directly in Supabase for tests

### DISCUSSION THREAD

**Discussion by n-d-r-d-g** [supabase/supabase#19468] [Feature Requests]
# Bug report

## Describe the bug

I'm currently using Twilio as phone provider and it works well for sending OTP via SMS. However, every OTP sent via SMS costs money and it doesn't make sense to spend credit during the testing phase. Is there a way to generate the OTP for a user via the Supabase dashboard, e.g. the SQL Editor?

## Additional context

The users will be testing the app I'm currently building in staging. Before asking the client to subscribe to a paid Twilio plan, I want to make sure there is a way to reduce costs, i.e. the client is only charged for SMS in production but not staging. Since it's a pay-as-you-go plan, I fear the costs will get pretty high during stress tests.

**Comment by biomassives** [comment]
seems like this is a <label>  Feature

**Comment by shawnmclean** [comment]
Has anyone been able to figure out a workaround?

Twilio's [article](https://www.twilio.com/blog/test-verify-no-rate-limits) showed:

```
Unfortunately, we do not offer a free testing sandbox since we have to pay underlying carrier fees for every message sent on our platform and Twilio's [test credentials](https://support.twilio.com/hc/en-us/articles/223183808-Why-aren-t-my-test-credentials-working-) are not compatible with the Verify API.
```
